### ISSUE TITLE
Should the Reflection function numFields() be parenless?

### ISSUE THREAD

**Issue by dlongnecke-cray** [chapel-lang/chapel#17985]
Should the Reflection function numFields() be parenless?

This depends on #17984.

Many times throughout the standard modules or builtin types, methods
are made parenless when they seem more like an attribute of a type
than a method on it.

Some examples would be:

  - `range.stride`
  - `domain.rank`
  - `numLocales` (Actually a `config const`, but still)

Should `numFields()` follow suit and drop the parens? It seems like a
good candidate as a statically computed value.

And while we are here, are there any other methods in the Reflection
module that could be considered candidates for the same?

**Comment by lydia-duncan** [comment]
This depends on if it is changed into a method (#17984), right?

**Comment by dlongnecke-cray** [comment]
It suuuuure does! Didn't really think that one through xD.

I will not close this since it's already open, but I will point away towards #17984.

**Comment by lydia-duncan** [comment]
It similarly seemed like we won't do this, closing
